pyproject.toml

```diff
@@ -1,3 +1,157 @@
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3 :: Only",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Mathematics",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Software Development",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+requires-python = ">=3.7"
+dependencies = [
+  "alembic>=1.5.0",
+  "cmaes>=0.9.0",
+  "colorlog",
+  "numpy",
+  "packaging>=20.0",
+  "scipy>=1.7.0",
+  "sqlalchemy>=1.3.0",
+  "tqdm",
+  "PyYAML",  # Only used in `optuna/cli.py`.
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+benchmark = [
+  "asv>=0.5.0",
+  "botorch",
+  "cma",
+  "scikit-optimize",
+  "virtualenv"
+]
+checking = [
+    "black",
+    "blackdoc",
+    "hacking",
+    "isort",
+    "mypy",
+    "types-PyYAML",
+    "types-redis",
+    "types-setuptools",
+    "typing_extensions>=3.10.0.0",
+]
+document = [
+    "cma",
+    "distributed",
+    "fvcore",
+    "lightgbm",
+    "matplotlib!=3.6.0",
+    "mlflow",
+    "pandas",
+    "pillow",
+    "plotly>=4.0.0",  # optuna/visualization.
+    "scikit-learn",
+    "scikit-optimize",
+    "sphinx",
+    "sphinx-copybutton",
+    "sphinx-gallery",
+    "sphinx-plotly-directive",
+    "sphinx_rtd_theme",
+    "torch==1.11.0",
+    "torchaudio==0.11.0",
+    "torchvision==0.12.0",
+]
+integration = [
+    "allennlp>=2.2.0",
+    # TODO(c-bata): Remove cached-path after allennllp supports v1.1.3
+    "cached-path<=1.1.2",
+    "botorch>=0.4.0",
+    "catalyst>=21.3",
+    "catboost>=0.26",
+    "chainer>=5.0.0",
+    "cma",
+    "distributed",
+    "fastai",
+    "lightgbm",
+    "mlflow",
+    "mpi4py",
+    "mxnet",
+    "pandas",
+    "pytorch-ignite",
+    "pytorch-lightning>=1.5.0",
+    "scikit-learn>=0.24.2",
+    "scikit-optimize",
+    "shap",
+    "skorch",
+    "tensorflow",
+    "tensorflow-datasets",
+    "torch==1.11.0",
+    "torchaudio==0.11.0",
+    "torchvision==0.12.0",
+    "wandb",
+    "xgboost",
+]
+optional = [
+    "matplotlib!=3.6.0",  # optuna/visualization/matplotlib
+    "pandas",  # optuna/study.py
+    "plotly>=4.0.0",  # optuna/visualization.
+    "redis",  # optuna/storages/redis.py.
+    "scikit-learn>=0.24.2",
+    # optuna/visualization/param_importances.py.
+]
+test = [
+    "codecov",
+    "fakeredis[lua]",
+    "kaleido",
+    "pytest",
+]
+
+[project.urls]
+homepage = "https://optuna.org/"
+repository = "https://github.com/optuna/optuna"
+documentation = "https://optuna.readthedocs.io"
+bugtracker = "https://github.com/optuna/optuna/issues"
```

Comment:
Will the names of the links on the PyPI page be the same?

Reply:
I think Yes.

Specifically, these names are seemingly mapped to links in PyPI by the following rules:
https://github.com/pypi/warehouse/blob/cd9dd933343f855aa8a3a8235c9baf46b0421249/warehouse/templates/packaging/detail.html#L20-L60

In my understanding, there is no naming rule for the project-url except to the maximum size of characters.
https://packaging.python.org/en/latest/specifications/core-metadata/#project-url-multiple-use

And PyPI shows some special links whose names are matched in the above `detail.html` rules.

🔗 https://ymd_h.gitlab.io/ymd_blog/posts/custom_link_pypi/